File: backend/FinanceApp.Domain/Entities/AuditLog.cs
namespace FinanceApp.Domain.Entities;

public class AuditLog : BaseEntity
{
    public long? UserId { get; set; }
    public string Action { get; set; } = string.Empty;
    public string EntityType { get; set; } = string.Empty;
    public long? EntityId { get; set; }
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
    public string? IpAddress { get; set; }
    public string? UserAgent { get; set; }

    // Navigation properties
    public User? User { get; set; }
}

File: frontend/src/features/system/types/auditLog.ts
// 审计日志
export interface AuditLog {
  id: number
  entityType: string
  entityId: number | null
  action: string
  oldValue?: string
  newValue?: string
  userId: number | null
  operatorName: string
  ipAddress: string
  createdAt: string
}

// 审计日志查询参数
export interface AuditLogQueryParams {
  page: number
  pageSize: number
  userId?: number
  username?: string
  action?: string
  entityType?: string
  startDate?: string
  endDate?: string
  sortBy?: string
  sortDesc?: boolean
}
